### ISSUE TITLE
await expect - new false positive in 1.8.0

### ISSUE THREAD

**Issue by dylang** [playwright-community/eslint-plugin-playwright#323]
Hi, it looks like the work done in https://github.com/playwright-community/eslint-plugin-playwright/commit/e283efc8e34a55f446268b11f0cfbbec310d7b2e is not correctly supporting async matchers. 

For example, this is now considered incorrect usage of `await`:

```ts
await expect(pageObject.tabTo('.first', '.third', { maxCount: 1 })).rejects.toThrow(
  'Could not reach the selector .third using the keyboard'
);

**Comment by mdmower** [comment]
I'm similarly seeing what I believe are false positives for `expect.poll()`:

```ts
await expect.poll(() => getSlowStorageValue(page, 'key')).toBe('value');
```

Documentation implies `await` is needed: https://playwright.dev/docs/test-assertions#expectpoll
